### PR TITLE
HIVE-28494: Iceberg: mvn build enables iceberg module by default

### PR DESCRIPTION
**Pull request by zhangbutao**
### Why are the changes needed?

[HIVE-25027](https://issues.apache.org/jira/browse/HIVE-25027) hidden the iceberg module by default. IMO, we have put lots of effort into iceberg module and it is more stable than before. We should enable the iceberg module by default in case of mvn build.

### Does this PR introduce _any_ user-facing change?

No
### Is the change a dependency upgrade?

No

### How was this patch tested?

After this change, you can build the package including iceberg moudle with this cmd:
`mvn clean install -DskipTests -Pdist`

### CODE REVIEW

**Review by deniskuzZ — APPROVED** [review]
LGTM +1

File: pom.xml [hunk 1]
```diff
@@ -2099,12 +2100,6 @@
         <module>itests</module>
       </modules>
     </profile>
-    <profile>
```
Comment (Indhumathi27) [inline pom.xml:2102]: We have Iceberg profile in itests module https://github.com/apache/hive/blob/3f6f940af3f60cc28834268e5d7f5612e3b13c30/itests/pom.xml#L501 as well which runs tests in qtest-iceberg module. This also should be enabled by default

Comment (zhangbutao) [reply pom.xml:2102]: Good catch! I checked the original ticket HIVE-25088.
Fixed in https://github.com/apache/hive/pull/5428/commits/8c1b26b13cd894085c2fab0a5d1b62c74432b757

Please help to check if this change is ok.
Thanks @Indhumathi27

**Review by zabetak — COMMENTED** [review]
In the past (HIVE-25998), this was breaking the compilation of the project in IntelliJ. Are these problems resolved?

**Comment by zhangbutao** [comment]
> In the past ([HIVE-25998](https://issues.apache.org/jira/browse/HIVE-25998)), this was breaking the compilation of the project in IntelliJ. Are these problems resolved?

@zabetak IMO, before HIVE-25998, we need to shade&patch lots of Apache Iceberg API&Core code by two patched moudle `patched-iceberg-api `& `patched-iceberg-core`, to make our hive-iceberg handler development more faster, and these shaded code can't be read easily by developers unless they manually add the missing dependency in the Intellij idea dependency moule. 

But now, the two patched moudle `patched-iceberg-api `& `patched-iceberg-core` are rarely used, and hive-iceberg handler module is more stable than before. So we can safely enable the iceberg module by default.

**Comment by zhangbutao** [comment]
> It seems that you also need to update the following file: `.github/workflows/docker-GA-images.yml`

Fixed in https://github.com/apache/hive/pull/5428/commits/a2735295654d16d6f9a95b66e63fb2381a4f1426
@zabetak Thanks a lot!

**Review by zabetak — APPROVED** [review]

**Pull request merged**